lib/linguist/heuristics.yml

```diff
@@ -157,6 +157,19 @@ disambiguations:
     pattern: '# Microsoft Developer Studio Generated Build File'
   - language: Faust
     pattern: '\bprocess\s*[(=]|\b(library|import)\s*\(\s*"|\bdeclare\s+(name|version|author|copyright|license)\s+"'
+- extensions: ['.e']
+  rules:
+  - language: E
+    pattern:
+    - '^\s*(def|var)\s+(.+):='
+    - '^\s*(def|to)\s+(\w+)(\(.+\))?\s+{'
+    - '^\s*(when)\s+(\(.+\))\s+->\s+{'
+  - language: Eiffel
+    pattern:
+    - '^\s*{/\w+'
```

Comment:
This pattern conflicts with the `E` heuristics above. Consider:

~~~E
def foo()
{

}
~~~

Remember, `\s` will match newlines in addition to tabs and spaces. In fact, if matching horizontal whitespace is your intention, it's much better to be explicit (in which case, replace `\s` with `[ \t]` where applicable).

IMHO, this pattern looks incredibly open-ended; such patterns have a high chance of matching something they weren't supposed to. I recommend deleting it, actually.

Reply:
I think I added this pattern because of a weird side-effect I was seeing when testing linguist against some random E, Eiffel, and Elixir projects. Mainly [this file](https://github.com/EiffelWebFramework/EWF/blob/master/tools/estudio_wizard/rootdir/resources/_vars.e) in EWF which linguist thinks is *Euphoria* but it actually looks like a template. I guess it's already misidentified if linguist thinks it's Eiffel when it's not.

I'm not too familiar with these other languages over all and my goal is just to get Euphoria identified without interfering with any other languages. To date, it seems the heuristics for E, Eiffel, and Elixir in linguist have been mostly automatic but since we're sharing the `.e` and `.ex` extensions, I was running into a lot of false positives early in my testing.

I can remove the pattern since it's a fringe case anyway. Aside from that case I think the heuristics are working well.

Reply:
Yes, please do remove it. An inappropriate heuristic can be much worse than no heuristic at all.